localstack-core/localstack/services/apigateway/next_gen/execute_api/handlers/integration_request.py

```diff
@@ -80,3 +100,64 @@ def get_integration_request_data(
             context_variables=context.context_variables,
             stage_variables=context.stage_variables,
         )
+
+    def render_request_template_mapping(
+        self,
+        context: RestApiInvocationContext,
+        template: str,
+    ) -> tuple[str, ContextVarsRequestOverride]:
+        request: InvocationRequest = context.invocation_request
+        body = request.get("body", b"").decode()
```

Comment:
nit: I don't think we should decode the body before checking if there's a template, because then we need to re-encode it. 

Maybe we could do the decoding later, and always return the body as `bytes`, so we could return the body as is. Then, after the rendering in this function you can encode it.

Then we don't need to encode it back in the caller method?

Reply:
Oh, also, forgot to add: `to_str` and `to_bytes` utils take care of the default encoding, and are similar to calling `.decode()` and `.encode()`.

Reply:
Thanks for the tip.
Updated